web/src/hash_parser.ts

```diff
@@ -161,10 +164,11 @@ export function is_spectator_compatible(hash: string): boolean {
 
     if (main_hash === "narrow") {
         let hash_section = get_hash_section(hash);
+        const second_hash_section = get_nth_hash_section(hash, 2);
         if (hash_section.startsWith("-")) {
             hash_section = hash_section.slice(1);
         }
-        if (!is_an_allowed_web_public_narrow(hash_section)) {
+        if (!is_an_allowed_web_public_narrow(hash_section, second_hash_section)) {
             return false;
         }
         return true;
```

Comment:
This can be a follow-up PR, but I'm pretty sure that this logic should be a loop over operators after parsing the narrow, rather than ignoring everything after the first operator/operand pair.